api/base/filters.py

```diff
@@ -106,22 +130,47 @@ def query_params_to_odm_query(self, query_params):
             query = None
         return query
 
-    # Used so that that queries by _id will work
-    def convert_key(self, key):
-        key = key.strip()
-        if self.serializer_class._declared_fields[key].source:
-            return self.serializer_class._declared_fields[key].source
-        return key
 
-    # Used to convert string values from query params to Python booleans when necessary
-    def convert_value(self, value):
-        value = value.strip()
-        if value in self.TRUTHY:
-            return True
-        elif value in self.FALSY:
-            return False
-        # Convert me to current user's pk
-        elif value == 'me' and not self.request.user.is_anonymous():
-            return self.request.user.pk
+class ListFilterMixin(FilterMixin):
+    """View mixin that adds a get_queryset_from_request method which uses query params
+    of the form `filter[field_name]=value` to filter a list of objects.
+
+    Subclasses must define `get_default_queryset()`.
+
+    Serializers that want to restrict which fields are used for filtering need to have a variable called
+    filterable_fields which is a frozenset of strings representing the field names as they appear in the serialization.
+    """
+
+    def get_default_queryset(self):
+        raise NotImplementedError('Must define get_default_queryset')
+
+    def get_queryset_from_request(self):
+        default_queryset = self.get_default_queryset()
```

Comment:
This is only used if self.request.QUERY_PARAMs is falsy, so this can go in the `else` block.